pandas/tests/io/test_gbq.py

```diff
@@ -149,33 +150,28 @@ def mock_read_gbq(sql, **kwargs):
 
 @pytest.mark.single
 class TestToGBQIntegrationWithServiceAccountKeyPath:
-    @classmethod
-    def setup_class(cls):
-        # - GLOBAL CLASS FIXTURES -
-        # put here any instruction you want to execute only *ONCE* *BEFORE*
-        # executing *ALL* tests described below.
-
+    @pytest.fixture()
+    def gbq_dataset(self):
+        # Setup Dataset
         _skip_if_no_project_id()
         _skip_if_no_private_key_path()
 
-        cls.client = _get_client()
-        cls.dataset = cls.client.dataset(DATASET_ID + "1")
+        self.client = _get_client()
```

Comment:
so I would add in the DESTINATION_TABLE here as well; generate a random string name each time, you can yield the table name